bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/classic/BigtableHBaseClassicSettings.java

```diff
@@ -0,0 +1,396 @@
+  }
+
+  private void buildChannelOptions(BigtableOptions.Builder builder) {
+    String channelCountStr = configuration.get(BIGTABLE_DATA_CHANNEL_COUNT_KEY);
+    if (channelCountStr != null) {
+      builder.setDataChannelCount(Integer.parseInt(channelCountStr));
+    }
+
+    // This is primarily used by Dataflow where connections open and close often. This is a
+    // performance optimization that will reduce the cost to open connections.
+    String useCachedDataPoolStr = configuration.get(BIGTABLE_USE_CACHED_DATA_CHANNEL_POOL);
+    if (useCachedDataPoolStr != null) {
+      builder.setUseCachedDataPool(Boolean.parseBoolean(useCachedDataPoolStr));
+    }
+
+    // This information is in addition to bigtable-client-core version, and jdk version.
+    StringBuilder agentBuilder = new StringBuilder();
+    agentBuilder.append("hbase-").append(VersionInfo.getVersion());
+    String customUserAgent = configuration.get(CUSTOM_USER_AGENT_KEY);
+    if (customUserAgent != null) {
+      agentBuilder.append(',').append(customUserAgent);
+    }
+    builder.setUserAgent(agentBuilder.toString());
+  }
+
+  private void buildClientCallOptions(BigtableOptions.Builder OptionsBuilder) {
+    CallOptionsConfig.Builder callOptionsBuilder =
+        OptionsBuilder.getCallOptionsConfig().toBuilder();
+
+    String useTimeoutStr = configuration.get(BIGTABLE_USE_TIMEOUTS_KEY);
+    if (!isNullOrEmpty(useTimeoutStr)) {
+      callOptionsBuilder.setUseTimeout(Boolean.parseBoolean(useTimeoutStr));
+    }
+
+    String shortRpcTimeoutMs = configuration.get(BIGTABLE_RPC_TIMEOUT_MS_KEY);
+    if (!isNullOrEmpty(shortRpcTimeoutMs)) {
+      callOptionsBuilder.setShortRpcTimeoutMs(Integer.parseInt(shortRpcTimeoutMs));
+    }
+
+    String longTimeoutMs = configuration.get(BIGTABLE_LONG_RPC_TIMEOUT_MS_KEY);
+    if (!isNullOrEmpty(longTimeoutMs)) {
+      callOptionsBuilder.setLongRpcTimeoutMs(Integer.parseInt(longTimeoutMs));
+    }
+
+    String mutateRpcTimeoutMs = configuration.get(BIGTABLE_MUTATE_RPC_TIMEOUT_MS_KEY);
+    if (!isNullOrEmpty(mutateRpcTimeoutMs)) {
+      callOptionsBuilder.setMutateRpcTimeoutMs(Integer.parseInt(mutateRpcTimeoutMs));
+    }
+
+    String readRowsRpcTimeoutMs = configuration.get(BIGTABLE_READ_RPC_TIMEOUT_MS_KEY);
+    if (!isNullOrEmpty(readRowsRpcTimeoutMs)) {
+      callOptionsBuilder.setReadRowsRpcTimeoutMs(Integer.parseInt(readRowsRpcTimeoutMs));
+    }
+
+    OptionsBuilder.setCallOptionsConfig(callOptionsBuilder.build());
+  }
+
+  private void buildCredentialOptions(BigtableOptions.Builder builder)
+      throws FileNotFoundException {
+
+    // This preserves user defined Credentials
+    if (configuration instanceof BigtableExtendedConfiguration) {
+      Credentials credentials = ((BigtableExtendedConfiguration) configuration).getCredentials();
+      builder.setCredentialOptions(CredentialOptions.credential(credentials));
+
+      // Here default is true, if its false then none of the service account would be accounted.
```

Comment:
This comment is not true...
the default is that configuration doesn't have the key, so the value is null which parseBoolean will interpret as false. 

I wonder if we can just ignore BIGTABLE_SERVICE_ACCOUNT_JSON_VALUE_KEY ? and make this:

if (instanceof BigtableExtendedConfiguration) {
...
} else if (BIGTABLE_NULL_CREDENTIAL_ENABLE_KEY) {
...
} else if (BIGTABLE_SERVICE_ACCOUNT_JSON_VALUE_KEY) {
} else if (BIGTABLE_SERVICE_ACCOUNT_JSON_VALUE_KEY) {
} else if (BIGTABLE_SERVICE_ACCOUNT_EMAIL_KEY) {
} 

can you think of any negative reprecussions?

Reply:
I assume you meant this(Basically to ignore `BIGTABLE_SERVICE_ACCOUNT_JSON_VALUE_KEY`)
```java
    if(instance instanceof BigtableExtendedConfiguration) {
      ...
    } else if(BIGTABLE_NULL_CREDENTIAL_ENABLE_KEY) {
      ...
    } else if(BIGTABLE_SERVICE_ACCOUNT_JSON_KEYFILE_LOCATION_KEY){
      ...
    } else if(BIGTABLE_SERVICE_ACCOUNT_EMAIL_KEY){
      ...
    }
``` 

then with this a user will not be able to pass key-value pairs as a string within the configuration instance. Although we can always advise users to move towards providing key file location. I believe we should be fine.

Reply:
argh, sorry I mis copy/pasted
I meant can we ignore `BIGTABLE_USE_SERVICE_ACCOUNTS_KEY`, we should keep BIGTABLE_SERVICE_ACCOUNT_JSON_VALUE_KEY